src/apps/competitions/tasks.py

```diff
@@ -94,6 +96,30 @@ class CompetitionUnpackingException(Exception):
     pass
 
 
+def get_data_key(obj, file_type, temp_directory, creator):
+    file_name = obj.get(file_type)
+    if not file_name:
+        return
+
+    file_path = os.path.join(temp_directory, file_name)
+    if os.path.exists(file_path):
+        new_dataset = Data(
+            created_by=creator,
+            type=file_type,
+            name=f"{file_type} @ {now().strftime('%m-%d-%Y %H:%M')}",
+            was_created_by_competition=True,
+        )
+        new_dataset.data_file.save(os.path.basename(file_path), File(open(file_path, 'rb')))
+        return new_dataset.key
+    elif len(file_name) in (32, 36):
+        # UUID are 32 or 36 characters long
+        # TODO send error message if invalid UUID or invalid filename
```

Comment:
should we do more here? Can make `django-unique-uuidfield` library !?

Reply:
I don't think this is a matter of uniqueness, this TODO is more about a filename that is 32 characters long, but the file isn't present. It treats it like a UUID but then doesn't raise any error if it can't find that UUID in the database anywhere. However, I would also love to write a `django-unique-uuidfield` package, since that is not something I have ever attempted before.

Reply:
Ah, kk, can track that here #119